src/server/api/login/handlers.ts

```diff
@@ -58,32 +60,35 @@ export async function generateOtp(req: Express.Request, res: Express.Response) {
 
     try {
       await setOtpForEmail(email, otpObject)
-      // Email out the otp (nodemailer)
-      mailOTP(email, otp, (mailError: Error) => {
-        if (!mailError) {
-          res.ok(jsonMessage('OTP generated and sent.'))
-        } else if (DEV_ENV) {
-          logger.warn('Allowing user to OTP even though mail errored.')
-          logger.warn(
-            'This may be an issue with your IP. More information can be found at https://support.google.com/mail/answer/10336?hl=en)',
-          )
-          logger.warn('This message should NEVER be seen in production.')
-          res.ok(jsonMessage('Error mailing OTP.'))
-        } else {
-          res.serverError(
-            jsonMessage('Error mailing OTP, please try again later.'),
-          )
-        }
-      })
     } catch (saveError) {
       res.serverError(jsonMessage('Could not save OTP hash.'))
       logger.error(`Could not save OTP hash:\t${saveError}`)
       return
     }
+    // Email out the otp (nodemailer)
+    try {
+      await mailOTP(email, otp)
+    } catch (error) {
+      if (DEV_ENV) {
```

Comment:
One idea that came to mind when reviewing this region would be that this could be circumvented using conditional binding in `inversify config` such that if DEV_ENV is `development`, bind the mailing interface to no-op lambdas instead of the original dependencies. That way `initMailer` and `mailOtp` become blank functions without the need to do this control flow